docs/tutorials/ruby-driver-create-client.txt

```diff
@@ -1129,55 +1129,102 @@ Note that when creating a client using the `block syntax <#block-syntax>`_ descr
+and cease background monitoring:
 
 .. code-block:: ruby
 
-    after_fork do |server, worker|
-      $client.close
-      $client.reconnect
-    end
+  client.reconnect
+
+.. note::
+
+  If the parent process performs operations on the Mongo client and does not
+  close it, the parent process will continue consuming a connection slot
+  in the cluster and will continue monitoring the cluster for as long as the
+  parent remains alive.
+
+Reconnecting Client Instances
+`````````````````````````````
+
+When the Ruby driver is used in a web application, it is recommended to not
+create any ``Mongo::Client`` instances in the management processes (prior to
+the workers being forked), and instead only create client instances in the
+workers.
 
-The above code assumes your Mongo::Client instance is stored in the $client
-global variable. It also keeps the MongoDB connection in the parent process
-around, and this connection will continue to consume resources such as
-a connection slot. If the parent process performs one time operation(s) on
-MongoDB and does not need its MongoDB connection once workers are forked,
-the following code will close the connection in the parent:
+It is possible, although not recommended, to use the same ``Mongo::Client``
+instances in parent and child processes. In order to do so, the instance
+must be closed and reconnected in the child process so that the background
+threads can be recreated:
 
 .. code-block:: ruby
 
-    before_fork do |server, worker|
-      $client.close
-    end
+  client.close
+  client.reconnect
 
-    after_fork do |server, worker|
-      $client.reconnect
-    end
+.. note::
+
+  This pattern should be used with Ruby driver version 2.6.2 or higher.
+  Previous driver versions did not recreate monitoring threads when
+  reconnecting.
 
-*Note:* This pattern should be used with Ruby driver version 2.6.2 or higher.
-Previous driver versions did not recreate monitoring threads when reconnecting.
+.. note::
 
-*Note:* If the parent process performs operations on the Mongo client and
-does not close it, the parent process will continue consuming a connection slot
-in the cluster and will continue monitoring the cluster for as long as the
-parent remains alive.
+  When closing and reconnecting the client instance in the child,
+  due to file descriptor sharing, the parent process may experience network
+  and monitoring errors.
+
+Web servers generally provide hooks that can be used by applications to
+perform actions when the worker processes are forked. The recommended hooks
+to use are:
+
+- For Puma, ``before_fork`` to close clients in the parent process and
+  ``on_worker_boot`` to reconnect in the child processes.
+- For Unicorn, ``before_fork`` to close clients in the parent process and
+  ``after_fork`` to reconnect clients in the child processes.
+- For Passenger, ``starting_worker_process`` to reconnect clients in the
+  child processes (Passenger does not appear to have a pre-fork hook).
```

Comment:
Is it worth linking to the documentation for each hook?

Reply:
I agree that might be helpful